src/librustdoc/test.rs

```diff
@@ -251,7 +251,9 @@ fn run_test(
     let mut compiler = Command::new(std::env::current_exe().unwrap().with_file_name("rustc"));
     compiler.arg("--crate-type").arg("bin");
     for cfg in &options.cfgs {
-        compiler.arg("--cfg").arg(&cfg);
+        if cfg != "rustdoc" {
```

Comment:
This also feels like the wrong place :)

I would expect us to not add the rustdoc cfg ourselves -- looking at it, I guess [here](https://github.com/rust-lang/rust/blob/master/src/librustdoc/config.rs#L347) -- not remove it later on. We don't want to remove it if the user has passed it.

Reply:
That's a good point I didn't think about... Checking more globally why we even need to do this. However, I think we need it at some point to check for the documentation build (not the tests build). I'm looking around to confirm it.